core/class.go

```diff
@@ -112,3 +115,76 @@ func flattenSierraEntryPoints(entryPoints []SierraEntryPoint) []*felt.Felt {
 	}
 	return result
 }
+
+func cairoClassHash(class Class) (*felt.Felt, error) {
+	switch t := class.(type) {
+	case *Cairo0Class:
+		return t.Hash(), nil
+	case *Cairo1Class:
+		return t.Hash(), nil
+	default:
+		return nil, errors.New("unknown class")
+	}
+}
+
+type CantVerifyClassHashError struct {
+	c           Class
+	hashFailure error
+	next        *CantVerifyClassHashError
+}
+
+func (e CantVerifyClassHashError) Unwrap() error {
+	if e.next != nil {
+		return *e.next
+	}
+	return nil
+}
+
+func (e CantVerifyClassHashError) Error() string {
+	return fmt.Sprintf("cannot verify class hash: %s", e.hashFailure)
+}
```

Comment:
With this change, if we pass in something like this `CantVerifyClassHashError{}`, we'll have `cannot verify class hash: %!s(<nil>)`. Is that okay?

@IronGauntlets

Reply:
Yes because we an in complete control of creating `CantVerifyClassHashError` and no one outside of the package should create this error. Therefore, any and all instances of `CantVerifyClassHashError` will habe non-nill `hashFailure`.